doc/advancedFeatures.rst

```diff
@@ -0,0 +1,43 @@
+Advanced Features
+=================
+.. Gradient Evaluation with Complex Step
+.. -------------------------------------
+
+.. Parallel Execution
+.. ------------------
+
+.. Storing Optimization History
+.. ----------------------------
+
+Hot start
+---------
+Hot start refers to the way optimizations are initialized.
+Suppose you run an optimization, and it was accidentally terminated prematurely, for example due to an iteration limit which was too low.
+If you restarted the optimization from the last iteration, you will likely incur a performance penalty, and end up taking far more steps than it would've taken if the original optimization was allowed to progress.
+This is because for many optimizers, the next iterate does not depend only on the current iterate, but effectively all the previous iterates as well.
+In a sense, these optimizers are path-dependent, and when you restart the optimization you lose all the accumulated history.
+
+Hot starting optimizations is a way to address this issue.
+pyOptSparse has the ability to read in the previous optimization history file, and `replay` the entire history starting at the original design variables, feeding ``funcs`` to the optimizer each time.
+For a deterministic optimizer, if all solver settings remain the same, then it should request the same sequence of iterates (up to machine precision), and if those have been previously evaluated, they are read from the history file and passed to the optimizer.
+If at any point the requested design variables diverge from the history, then we will actually perform a function evaluation, and all subsequent points will be newly evaluated.
+For this process to work, the following must be true:
+
+-  The optimizer is deterministic.
+   Note that a stochastic optimizer can still be made deterministic even if it uses random numbers, as long as the random seed is fixed.
+   If the optimizer performs a random gradient check (e.g. SNOPT), it's best to disable these just in case.
+-  Optimizer settings that affect the path of the optimization must remain the same.
+   It is perfectly fine to change for example settings related to print outs, but not those affecting the line search.
+-  The hot start file is itself cold started.
+   It is not possible to hot start from another hot start.
```

Comment:
Can you expand on this a bit? From this it sounds like that you can hot-start an optimization only once, but I am pretty sure this is not the case, right?

Reply:
Oh shoot my mistake, you are right I think I just confused myself there. Because the history file contains the full history including the previous history file, it's fine to hotstart multiple times. I will fix this.